docs/resources/dds_database_role.md

```diff
@@ -0,0 +1,93 @@
+---
+subcategory: "Document Database Service (DDS)"
+---
+
+# huaweicloud_dds_database_role
+
+Manages a database role resource within HuaweiCloud.
+
+## Creating a database role under a DDS instance
+
+```hcl
+variable "instance_id" {}
+variable "role_name" {}
+variable "owned_role_name" {}
+
+resource "huaweicloud_dds_database_role" "test" {
+  instance_id = var.instance_id
+
+  name = var.role_name
+
+  roles {
+    name = var.owned_role_name
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region where the DDS instance is located.
+  Changing this parameter will create a new role.
+
+* `instance_id` - (Required, String, ForceNew) Specifies the DDS instance ID to which the role belongs.
+  Changing this parameter will create a new role.
+
+* `name` - (Required, String, ForceNew) Specifies the role name.
+  The name can contain `1` to `64` characters, only letters, digits, underscores (_), hyphens (-) and dots (.) are
+  allowed. Changing this parameter will create a new role.
+
+* `db_name` - (Optional, String, ForceNew) Specifies the database name to which the role belongs. Defaults to **admin**.
+  The name can contain `1` to `64` characters, only letters, digits and underscores (_) are allowed.
+  Changing this parameter will create a new role.
+
+* `roles` - (Optional, List, ForceNew) Specifies the list of roles owned by the current role.
+  The [object](#dds_database_owned_roles) structure is documented below.
+  Changing this parameter will create a new role.
+
+<a name="dds_database_owned_roles"></a>
+The `roles` block supports:
+
+* `name` - (Required, String, ForceNew) Specifies the name of role owned by the current role.
+  The name can contain `1` to `64` characters, only letters, digits, underscores (_), hyphens (-) and dots (.) are
+  allowed. Changing this parameter will create a new role.
+
+* `db_name` - (Optional, String, ForceNew) Specifies the database name to which this owned role belongs.
```

Comment:
duplicate `db_name`, remove this one as the values must be same

Reply:
After testing, I found the role can have the role of other databases, so the `db_name` needs to be reserved.